code/rdb/datastripe.q

```diff
@@ -1,13 +1,18 @@
+.rdb.extendperiods:.ds.periodstokeep
 .rdb.datastripeendofperiod:{[currp;nextp;data]
 
     .lg.o[`reload;"reload command has been called remotely"];
 
     t:tables[`.] except .rdb.ignorelist;
 
+    // check if tail sort process is complete
+    if[.rdb.tailsortcomplete;.rdb.extendperiods:.ds.periodstokeep];
```

Comment:
Is this not reassigned on line 1 regardless of tailsortcomplete?

Reply:
Line 1 is just initialising `.rdb.extendperiods` to whatever value `.ds.periodstokeep` is set to, it only runs when the code is first loaded